int32 device_id
int32 task_id
int32 map_id
int32 flag
---
bool success
string message